### PR TITLE
Only incude single solver .so dependency in seperate solver packages

### PR DESCRIPTION
**Pull request by gasper-vrhovsek**
This commit will fix building OpenFOAM packages by leaving only the solver .so file that corresponds to the package being built.

`openfoam_core.patch` has been moved to the `common` directory, because it is used for all solver packages. Additionally `reconstructPar` and `decomposePar` links are removed from specific solver packages, but left in `openfoam.core` package.

@miha-plesko @gberginc please review

### CODE REVIEW

File: docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh [hunk 1]
```diff
@@ -16,9 +16,15 @@ echo "/usr/bin/pimpleFoam.so: ${OSV_DIR}/mike-apps/OpenFOAM/ROOTFS/usr/bin/pimpl
 
 # Patch the module file to exclude the dependency on HTTP server
 cd ${OSV_DIR}
+patch -p1 < ${COMMON_DIR}/openfoam_core.patch
 patch -p1 < ${COMMON_DIR}/openfoam_remove_ompi.patch
 ${OSV_DIR}/scripts/build image=OpenFOAM export=all usrskel=none export_dir=$PACKAGE_RESULT_DIR -j ${CPU_COUNT}
 
+# Add simpleFoam solver
+cp ${OSV_DIR}/mike-apps/OpenFOAM/ROOTFS/usr/bin/pimpleFoam.so ${PACKAGE_RESULT_DIR}/usr/bin
+rm ${PACKAGE_RESULT_DIR}/usr/bin/decomposePar
```
Comment (gberginc) [inline docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh]: Looking at the [core script](https://github.com/gasper-vrhovsek/capstan-packages/blob/c91a8ea373bfbd547b5e86e2497c7f11c58121b7/docker_files/recipes/openfoam.core-2.4.0/build.sh#L22-L23)  `decomposePar` and `reconstructPar` are not actually copied to `/usr/bin` be default.

Are there `rm`s mandatory?

**Review by gberginc — COMMENTED** [review]
@gasper-vrhovsek I apologise for missing this one! I have a minor question which, if my understanding is correct, could slightly simplify recipes for individual solvers.

If not, we can merge as it is.

@miha-plesko can you review as well?

**Comment by miha-plesko** [comment]
I didn't forget about this one, but would like to test it locally before merge. Right after I finish my part in 5.3 ;)

File: docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh (hunk 1 above)
Comment (gasper-vrhovsek) [reply docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh]: It weren't the `decomposePar.so` and `reconstructPar.so` that were the problems, but the symbolic links that get created and should point to those two files, but in this case pointed to nowhere.

Comment (gasper-vrhovsek) [reply docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh]: In the result logs these lines get printed out:
```
  940: [INFO] added link /result/intermediate/openfoam.pimplefoam-2.4.0/usr/bin/decomposePar -> decomposePar.so
  941  [INFO] added link /result/intermediate/openfoam.pimplefoam-2.4.0/usr/bin/reconstructPar -> reconstructPar.so
```
I'll try to find where this happens. If possible i'll implement a better solution.

Comment (gasper-vrhovsek) [reply docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh]: Thanks to @miha-plesko we found in which bush the rabbit is hiding:
https://github.com/mikelangelo-project/mike-apps/blob/master/OpenFOAM/GET#L132-L134

:)

Comment (miha-plesko) [reply docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh]: As discovered offline with @gasper-vrhovsek the problem is that [GET script](https://github.com/mikelangelo-project/mike-apps/blob/master/OpenFOAM/GET#L132-L134) of the mike-apps/OpenFOAM creates these two links. We agreed that it's best to fix the GET script and remove the `rm`s from here.

File: docker_files/recipes/openfoam.core-2.4.0/build.sh [hunk 2]
```diff
@@ -13,7 +13,7 @@ set -o errtrace
```
Comment (miha-plesko) [inline docker_files/recipes/openfoam.core-2.4.0/build.sh:13]: Whenever we modify build.sh, also package version should be shifted. Please update L29 from `0.3` to `0.4`

File: docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh [hunk 3]
```diff
@@ -16,9 +16,15 @@ echo "/usr/bin/pimpleFoam.so: ${OSV_DIR}/mike-apps/OpenFOAM/ROOTFS/usr/bin/pimpl
```
Comment (miha-plesko) [inline docker_files/recipes/openfoam.pimplefoam-2.4.0/build.sh:16]: Update version

File: docker_files/recipes/openfoam.pisofoam-2.4.0/build.sh [hunk 4]
```diff
@@ -16,9 +16,15 @@ echo "/usr/bin/pisoFoam.so: ${OSV_DIR}/mike-apps/OpenFOAM/ROOTFS/usr/bin/pisoFoa
```
Comment (miha-plesko) [inline docker_files/recipes/openfoam.pisofoam-2.4.0/build.sh:16]: Update version

File: docker_files/recipes/openfoam.poroussimplefoam-2.4.0/build.sh [hunk 5]
```diff
@@ -16,9 +16,15 @@ echo "/usr/bin/porousSimpleFoam.so: ${OSV_DIR}/mike-apps/OpenFOAM/ROOTFS/usr/bin
```
Comment (miha-plesko) [inline docker_files/recipes/openfoam.poroussimplefoam-2.4.0/build.sh:16]: Version

**Review by miha-plesko — CHANGES_REQUESTED** [review]
Apologies for insane delay @gasper-vrhovsek . Please see the comments below. Also, as discussed on Slack, please use

```
$ scripts/build image=OpenFOAM export=none
```

in solver packages and then manually cherry-pick just the one .so file you need.

**Comment by gasper-vrhovsek** [comment]
@miha-plesko please review the most recent changes. I have implemented the changes you requested and tested building openfoam packages and everything looked ok.

**Review by miha-plesko — APPROVED** [review]
Thanks it looks great now! Merging...

**Pull request merged**